packages/attune/src/server/repo/dist/delete.rs

```diff
@@ -0,0 +1,178 @@
+    .unwrap()
+    .ok_or_else(|| {
+        ErrorResponse::builder()
+            .status(axum::http::StatusCode::NOT_FOUND)
+            .error("REPO_NOT_FOUND")
+            .message("repository not found")
+            .build()
+    })?;
+
+    // Query components and their architectures for S3 cleanup
+    let components = sqlx::query!(
+        r#"
+        SELECT
+            c.name as component_name,
+            pi.architecture::text as architecture
+        FROM debian_repository_release r
+        JOIN debian_repository_component c ON c.release_id = r.id
+        JOIN debian_repository_index_packages pi ON pi.component_id = c.id
+        WHERE r.repository_id = $1 AND r.distribution = $2
+        "#,
+        repo.id,
+        distribution_name,
+    )
+    .fetch_all(&mut *tx)
+    .await
+    .unwrap();
+
+    // Cascade will handle related records when deleting the distribution.
+    let result = sqlx::query!(
+        r#"
+        DELETE FROM debian_repository_release
+        WHERE repository_id = $1 AND distribution = $2
+        "#,
+        repo.id,
+        distribution_name,
+    )
+    .execute(&mut *tx)
+    .await
+    .unwrap();
+
+    // If no rows were affected, the distribution was already deleted.
+    // We return an OK response here because the user's goal was accomplished regardless.
+    if result.rows_affected() == 0 {
+        return Ok(Json(DeleteDistributionResponse::default()));
+    }
+
+    // Find and delete orphaned packages; the returning clause is for S3 cleanup.
+    let orphaned = sqlx::query!(
+        r#"
+        DELETE FROM debian_repository_package p
+        WHERE p.tenant_id = $1
+        AND NOT EXISTS (
+            SELECT 1 FROM debian_repository_component_package cp
+            WHERE cp.package_id = p.id
+        )
+        RETURNING p.id, p.s3_bucket, p.sha256sum
+        "#,
+        tenant_id.0,
+    )
+    .fetch_all(&mut *tx)
+    .await
+    .unwrap();
+
+    // Database state is correct, so we can commit the transaction.
+    // Now all we need to do is clean up S3 objects.
+    tx.commit().await.unwrap();
+
+    // Clean up S3 objects for this distribution based on known paths.
+    let prefix = format!("{}/dists/{}", repo.s3_prefix, distribution_name);
+    let keys = {
+        // Deletes distribution metadata files.
+        let mut keys = vec![
+            format!("{prefix}/Release"),
+            format!("{prefix}/Release.gpg"),
+            format!("{prefix}/InRelease"),
+        ];
+
+        // Deletes component metadata files.
+        keys.extend(components.iter().flat_map(|record| {
+            let prefix = format!(
+                "{}/{}/binary-{}",
+                prefix,
+                record.component_name,
+                record.architecture.as_ref().unwrap().replace('_', "-")
+            );
+            // TODO: When compressed indexes are implemented, add their deletion here.
+            [format!("{prefix}/Packages"), format!("{prefix}/Release")]
```

Comment:
[likely bug] Each individual component does not have a `Release` file, only a `Packages` index. 

Did you run this code path? I'm surprised you didn't get a 404 on the `DeleteObject`.

Reply:
Yes, this did work without any errors- but good point, I've changed it to only remove `.../Packages`